src/utils/changeChain.ts

```diff
@@ -0,0 +1,51 @@
+import {
+  internalAddressesPerNetwork,
+  ValidNetworkNameTypes,
+} from "../utils/contracts/contractAddresses/internalAddresses";
+import { injectedConnector } from "../hooks/injectedConnectors";
+
+function switchEthereumChain(connector: any, chainId: number) {
```

Comment:
Use of `any` will not result in merges. Also, please annotate your return types.

Reply:
So actually the `connector` name is an error it should be `provider` since I'm getting it from `injectedConnector.getProvider()` which specifies the return type to `Promise<any>`, but let me see what can I do about it...

Reply:
Maybe I can prevent sending the provider as a parameter and I can do something like this:
```
  try {
    await provider.request(switchEthereumChainParams(chainId))
  } catch {
    await provider.request(addEthereumChainParams(chain))
  }
```

Reply:
I'm making this change to that
```
injectedConnector.getProvider().then(async (provider: any) => {
  await provider
    .request(createRequestArguments("wallet_switchEthereumChain", chain))
    .catch(
      provider.request(
        createRequestArguments("wallet_addEthereumChain", chain)
      )
    );
});
```

Reply:
I'll need to do some work on this myself; I'm writing a commit and pushing it to your PR to assist, since it's fairly technical.